application/controllers/template_controller.py

```diff
@@ -18,7 +18,7 @@ def get_template_by_id(template_id):
     try:
         template = db.session.query(CommunicationTemplate).filter(CommunicationTemplate.id == template_id).first()
```

Comment:
Does db.session.query.first() actually throw a SQLAlchemyError. I couldn't find it in the docs only on the commit

Reply:
SQLAlchemyError is the base SA exception, and could be thrown here regardless of a missing object returned from the WHERE. 

Perhaps @benjefferies is right in that I also can't find anything to suggest that `first()` returns anything other than `None` or the object but this is then checked against when `_get_template_by_id` is called ([example](https://github.com/ONSdigital/rm-comms-template-service/pull/3/files#diff-48196c61fe5423823ad87d5f6a94cf8dR54)).